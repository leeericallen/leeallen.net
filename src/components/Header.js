import { Link } from 'gatsby';
import React from 'react';

import logo from '../assets/icons/logo.svg';
import resume from '../downloads/lee-allen-resume.pdf';
import styles from './Header.module.scss';

const NavItem = props => {
  const itemClassNames = [
    styles.navItem,
    `typography-semi-bold`,
    `typography-small`,
  ];

  if (props.pathname.match(props.pattern)) itemClassNames.push(styles.navItemActive);

  return (
    <li className={itemClassNames.join(` `)}>
      <Link className={styles.navLink} to={props.href}>{props.label}</Link>
    </li>
  );
};

export default props => {
  const navItems = [
    {
      href: `/`,
      label: `Home`,
      pattern: /^\/$/,
    },
    {
      href: `/work`,
      label: `Work`,
      pattern: /^\/work/,
    },
  ];
  const navLogoClassNames = [styles.navLogo];

  if (props.pathname === `/`) navLogoClassNames.push(styles.navLogoHome);

  return (
    <header className={styles.headerWrapper}>
      <div className={styles.header}>
        <Link className={navLogoClassNames.join(` `)} to={`/`}>
          <img
            src={logo}
            width={32}
            height={32}
            alt={`Logo`}/>
        </Link>
        {props.pathname !== `/` ? (
          <nav className={styles.nav}>
            <ol className={styles.navList}>
              {navItems.map((navItem, index) =>
                <NavItem key={index} {...navItem} pathname={props.pathname}/>)}
            </ol>
          </nav>
        ) : null}
      </div>
    </header>
  );
};
