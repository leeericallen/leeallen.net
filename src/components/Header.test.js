import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';

describe(`Header`, () =>
  it(`renders correctly on home page`, () => {
    const props = {
      pathname: `/`,
    };

    const tree = renderer.create(<Header pathname={`/`} />).toJSON();
    expect(tree).toMatchSnapshot();
  }));

describe(`Header`, () =>
  it(`renders correctly on Word page`, () => {
    const props = {
      pathname: `/`,
    };

    const tree = renderer.create(<Header pathname={`/word`} />).toJSON();
    expect(tree).toMatchSnapshot();
  }));
