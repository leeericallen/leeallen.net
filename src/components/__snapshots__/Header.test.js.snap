// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly on Word page 1`] = `
<header
  className="headerWrapper"
>
  <div
    className="header"
  >
    <a
      aria-current="page"
      className="navLogo"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      <img
        alt="Logo"
        height={32}
        src="test-file-stub"
        width={32}
      />
    </a>
    <nav
      className="nav"
    >
      <ol
        className="navList"
      >
        <li
          className="navItem typography-semi-bold typography-small"
        >
          <a
            aria-current="page"
            className="navLink"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="navItem typography-semi-bold typography-small"
        >
          <a
            className="navLink"
            href="/work"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Work
          </a>
        </li>
        <li
          className="navItem typography-semi-bold typography-small navItemActive"
        >
          <a
            className="navLink"
            href="/word"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Writing
          </a>
        </li>
      </ol>
    </nav>
  </div>
</header>
`;

exports[`Header renders correctly on home page 1`] = `
<header
  className="headerWrapper"
>
  <div
    className="header"
  >
    <a
      aria-current="page"
      className="navLogo navLogoHome"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      <img
        alt="Logo"
        height={32}
        src="test-file-stub"
        width={32}
      />
    </a>
  </div>
</header>
`;
